### ISSUE TITLE
Parameter of g in backward function is buggy.

### ISSUE THREAD

**Issue by YagaoDirac** [pytorch/pytorch#75312]
### 🐛 Describe the bug

`
import math 
import numpy
import torch

class SafeMul(torch.autograd.Function):
    @staticmethod
    def forward(ctx, x, y):
        return torch.ones_like(x)
    @staticmethod
    def backward(ctx, g):
        print(F"g before {g}")
        g[1] = 123.
        print(F"g after {g}")
        
        return #The return value has nothing to do with this report.
    
x = torch.tensor([1., 1., 1.], requires_grad = True)
y = torch.tensor([1., 1., 1.], requires_grad = True)
r = SafeMul.apply(x, y).sum()
r.backward()
`

The debug info looks like:
g before tensor([1., 1., 1.])
g after tensor([123., 123., 123.])

The expression g[1] = 123. modifies all the 3 elements. Is this intentional? Did I mess anything?

### Versions

pytorch 1.11.0
numpy 1.20.3
python 3.9.7

**Comment by YagaoDirac** [comment]
Ok, I tried mean() instead of sum(). It works... So the bug is probably in the sum() function.

**Comment by ngimel** [comment]
This is expected behavior, as your backward function is getting a discontiguous tensor as `g` if you compute loss as `sum`, see #64529. You need to first call `g=g.contiguous()` if you want to avoid it.
